nixos/modules/services/misc/matrix-synapse.nix

```diff
@@ -719,6 +719,9 @@ in {
           --config-path ${configFile} \
           --keys-directory ${cfg.dataDir} \
           --generate-keys
+
+        # restrict permissions on already existing signing keys
+        chmod 0600 ${cfg.dataDir}/homeserver.signing.key
```

Comment:
Maybe also on the media directory?

Reply:
Is it actually possible to serve the media without going through synapse? That would be very reasonable IMHO but unlikely as synapse will fetch them lazily from other servers if they are missing from the local copy.

Reply:
They might also be encrypted, so I think they should be served through synapse.

Reply:
Why? It is supposed to be end-to-end encryption. The server shouldn't even be able to decrypt them.

Reply:
I wonder if it encapsulates the message in some form; though I only see `blob:` scheme urls for images in element.

Reply:
There is https://github.com/turt2live/matrix-media-repo, so it's possible somehow, if I'm not misunderstanding things.

Reply:
Though that looks like a scenario where the media history is shared among multiple homeservers or even outsourced to an S3 backend. Either way we could set the media directory 750 and thereby require group access to read and access it. Does that sound good?